service/frontend/fx.go

```diff
@@ -124,11 +124,10 @@ var Module = fx.Options(
 	fx.Invoke(RegisterOpenAPIHTTPHandler),
 	fx.Provide(HTTPAPIServerProvider),
 	fx.Provide(NewServiceProvider),
-	fx.Provide(IncomingServiceClientProvider),
-	fx.Provide(IncomingServiceRegistryProvider),
-	fx.Provide(OutgoingServiceRegistryProvider),
+	fx.Provide(NexusEndpointClientProvider),
+	fx.Provide(NexusEndpointRegistryProvider),
 	fx.Invoke(ServiceLifetimeHooks),
-	fx.Invoke(IncomingServiceRegistryLifetimeHooks),
+	fx.Invoke(EndpointRegistryLifetimeHooks),
```

Comment:
Is this going to conflict with the providers / lifetime hooks in `components/nexusoperations/fx`? Should we consider having one registry provider, maybe at the `resource` fx level?

Reply:
It's not conflicting ATM.
This is still subject to change though, I'm leaving as is for now.